src/gtc/common.py

```diff
@@ -831,6 +831,27 @@ def is_single_index(self) -> bool:
 
         return abs(self.end.offset - self.start.offset) == 1
 
+    def overlap(self, other: "HorizontalInterval") -> bool:
```

Comment:
I would like to rename this `overlaps`, but since the GitLab CI has already run, we should do this as a future step next time this file is touched.

Reply:
udpated